[CIR] Build failure after StructType/UnionType split: stale `RecordType::get()` call in `CIRGenTypes.cpp`

Building `llvm-project@main` fails with compile errors:

```
/home/tenhian/github/forked/llvm-project/clang/lib/CIR/CodeGen/CIRGenTypes.cpp:646:37: error: no member named 'get' in 'cir::RecordType'
  646 |       resultType = cir::RecordType::get(&getMLIRContext(), /*members=*/elements,
      |                                     ^~~
/home/tenhian/github/forked/llvm-project/clang/lib/CIR/CodeGen/CIRGenTypes.cpp:648:67: error: no member named 'Struct' in 'cir::RecordType'; did you mean 'isStruct'?
  648 |                                         /*kind=*/cir::RecordType::Struct);
      |                                                                   ^~~~~~
      |                                                                   isStruct
/home/tenhian/github/forked/llvm-project/clang/include/clang/CIR/Dialect/IR/CIRTypes.h:117:8: note: 'isStruct' declared here
  117 |   bool isStruct() const;
      |        ^
/home/tenhian/github/forked/llvm-project/clang/lib/CIR/CodeGen/CIRGenTypes.cpp:648:67: error: call to non-static member function without an object argument
  648 |                                         /*kind=*/cir::RecordType::Struct);
      |                                                  ~~~~~~~~~~~~~~~~~^~~~~~
3 errors generated.
ninja: build stopped: subcommand failed.
```

## To reproduce

```
cmake -G Ninja \
  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
  -DLLVM_ENABLE_ASSERTIONS=ON \
  -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
  -DCMAKE_C_COMPILER=clang \
  -DCMAKE_CXX_COMPILER=clang++ \
  -DLLVM_USE_LINKER=lld \
  -DLLVM_TARGETS_TO_BUILD="X86" \
  -DLLVM_ENABLE_PROJECTS="clang;mlir" \
  -DCLANG_ENABLE_CIR=ON \
  -DBUILD_SHARED_LIBS=ON \
  -DLLVM_CCACHE_BUILD=ON \
  -DLLVM_BUILD_EXAMPLES=OFF \
  -DLLVM_INCLUDE_BENCHMARKS=OFF \
  -DLLVM_BUILD_TESTS=OFF \
  -DLLVM_BUILD_DOCS=OFF \
  -DCMAKE_INSTALL_PREFIX=$PWD/install \
  -S llvm \
  -B build-devel

ninja -C build-devel
```

## Root cause

PR #199790 (merged Jun 1) split the old tablegen-defined `cir::RecordType` into separate `cir::StructType` and `cir::UnionType` types, with `RecordType` becoming a hand-written view class. This removed the ODS-generated `RecordType::get()` static factory.

PR #200668 (merged Jun 6) was written against the pre-split API and landed with stale calls at `llvm-project/clang/lib/CIR/CodeGen/CIRGenTypes.cpp:646-648`:

```cpp
      resultType = cir::RecordType::get(&getMLIRContext(), /*members=*/elements,
                                        /*packed=*/false, /*padded=*/false,
                                        /*kind=*/cir::RecordType::Struct);
```

## Related

  - #199790 — Split RecordType into StructType/UnionType
  - #200668 — Initialization of atomic aggregates with padding (the commit that introduced the stale call)

## May useful

```cpp
      resultType = cir::StructType::get
```

--- a/clang/lib/CIR/CodeGen/CIRGenTypes.cpp
+++ b/clang/lib/CIR/CodeGen/CIRGenTypes.cpp
@@ -643,9 +643,9 @@ mlir::Type CIRGenTypes::convertType(QualType type) {
       auto paddingArray =
           cir::ArrayType::get(cgm.sInt8Ty, (atomicSize - valueSize) / 8);
       mlir::Type elements[] = {resultType, paddingArray};
-      resultType = cir::RecordType::get(&getMLIRContext(), /*members=*/elements,
+      resultType = cir::StructType::get(&getMLIRContext(), /*members=*/elements,
                                         /*packed=*/false, /*padded=*/false,
-                                        /*kind=*/cir::RecordType::Struct);
+                                        /*is_class=*/false);
     }
 
     break;

